chain/consensus/hierarchical/subnet/submgr/manager.go

```diff
@@ -548,6 +559,28 @@ func (s *Service) JoinSubnet(
 		return cid.Undef, err
 	}
 
+	// TODO: replace this with event listener
+	go func() {
+		var newst subnet.SubnetState
+
+		subnetAct, err := parentAPI.StateGetActor(ctx, SubnetActor, types.EmptyTSK)
+		if err != nil {
+			log.Infow("cannot get subnetAct for stats: %w", err)
+			return
+		}
+		pbs := blockstore.NewAPIBlockstore(parentAPI)
+		pcst := cbor.NewCborStore(pbs)
+		if err := pcst.Get(ctx, subnetAct.Head, &newst); err != nil {
+			log.Infow("getting actor state error after join: %w", err)
+			return
+		}
+
+		if (startStatus == subnet.Inactive || startStatus == subnet.Instantiated) &&
+			newst.Status == subnet.Active {
+			stats.Record(ctx, metrics.SubnetActiveCount.M(1))
```

Comment:
I think we really need the EventListener to do this in the right way. Instead of having to count up and down as a peer joins or leave a subnet, we could report the absolute number of active and killed subnets. Even if this approach is easier, I think it is worth exploring the event listener approach right away as it will give us all that we need to report certain metrics "real-time".